testing/integration/upgrade_standalone_same_commit_test.go

```diff
@@ -0,0 +1,381 @@
+		srcPackage, err := startFixture.SrcPackage(ctx)
+		require.NoErrorf(t, err, "error retrieving start fixture source package")
+
+		originalPackageFileName := filepath.Base(srcPackage)
+
+		// integration test fixtures and package names treat the version as a string including the "-SNAPSHOT" suffix
+		// while the repackage functions below separate version from the snapshot flag.
+		// Normally the early release versions are not snapshots but this test runs on PRs and main branch when we test
+		// starting from SNAPSHOT packages, so we have to work around the fact that we cannot simply re-generate the packages
+		// by defining versions in 2 separate ways for repackage hack and for fixtures
+		buildMetadataForAgentFixture := newVersionBuildMetadata
+		if currentVersion.IsSnapshot() {
+			buildMetadataForAgentFixture += "-SNAPSHOT"
+		}
+		versionForFixture := version.NewParsedSemVer(currentVersion.Major(), currentVersion.Minor(), currentVersion.Patch(), "", buildMetadataForAgentFixture)
+
+		// calculate the new package name
+		newPackageFileName := strings.Replace(originalPackageFileName, currentVersion.String(), versionForFixture.String(), 1)
+		t.Logf("originalPackageName: %q newPackageFileName: %q", originalPackageFileName, newPackageFileName)
+
+		newPackageContainingDir := t.TempDir()
+		newPackageAbsPath := filepath.Join(newPackageContainingDir, newPackageFileName)
+
+		// hack the package based on type
+		ext := filepath.Ext(originalPackageFileName)
+		if ext == ".gz" {
+			// fetch the next extension
+			ext = filepath.Ext(strings.TrimRight(originalPackageFileName, ext)) + ext
+		}
+		switch ext {
+		case ".zip":
+			t.Logf("file %q is a .zip package", originalPackageFileName)
+			repackageZipArchive(t, srcPackage, newPackageAbsPath, parsedNewVersion)
+		case ".tar.gz":
+			t.Logf("file %q is a .tar.gz package", originalPackageFileName)
+			repackageTarArchive(t, srcPackage, newPackageAbsPath, parsedNewVersion)
+		default:
+			t.Logf("unknown extension %q for package file %q ", ext, originalPackageFileName)
+			t.FailNow()
+		}
+
+		// Create hash file for the new package
+		err = mage.CreateSHA512File(newPackageAbsPath)
+		require.NoErrorf(t, err, "error creating .sha512 for file %q", newPackageAbsPath)
+
+		// I wish I could just pass the location of the package on disk to the whole upgrade tests/fixture/fetcher code
+		// but I would have to break too much code for that, when in Rome... add more code on top of inflexible code
+		repackagedLocalFetcher := atesting.LocalFetcher(newPackageContainingDir)
+
+		endFixture, err := atesting.NewFixture(t, versionForFixture.String(), atesting.WithFetcher(repackagedLocalFetcher))
+		require.NoErrorf(t, err, "error creating end fixture with LocalArtifactFetcher with dir %q", newPackageContainingDir)
+
+		err = upgradetest.PerformUpgrade(ctx, startFixture, endFixture, t,
+			upgradetest.WithUnprivileged(unprivilegedAvailable),
+			upgradetest.WithDisableHashCheck(true),
```

Comment:
Rather than disabling the check, could you invert it? That is require the hash to be the same?

That would enforce that the commit still reads the same after all of the package hacking that goes on as a condition of the test.

Reply:
This is to disable the check before starting the upgrade in our integration tests: it's the "fix" we have to avoid hash collisions then upgrading from a snapshot to another snapshot of the same version (it has been introduced by @rdner not too long ago)

The hash after upgrade is checked anyways using the endFixture which in our case has obviously the same hash